### PR TITLE
1.2.0

### PR DESCRIPTION
**Pull request by tetsuo**
* migrates all the way to [debug](https://github.com/visionmedia/debug) for logging
* exposes `debug` so it is possible to do `kd.debug.enable('kd:*')` in browser
* removes obsolete `appendToDOMBody`
* removes obsolete `SplitView` methods
* removes obsolete `deprecated` method in `view`
* log messages are redacted
* npm: removes lodash.throttle@2.4.1, adds lodash@3.6.0
* makes `dia/diascene` throttle indeed

### CODE REVIEW

**Comment by tetsuo** [comment]
a failed attempt to turn `object#bound` api to make use of `lodash#memoize` and `WeakMap` can be seen here: https://github.com/tetsuo/kd/commit/3a9a7717a860982b0af1520178a98fcb06337125

this required a tremendous effort of refactoring on _the other side,_ but unfortunately this is not only slower than the current implementation also requires some more tremendous effort of refactoring on _the other side._

**Comment by tetsuo** [comment]
i haven't been able to test extensively yet, just writing a reminder down here to give a reference from a commit.

is `kd.EventEmitter#off` broken? do you think we should make it obsolete and just inherit from [events.EventEmitter](https://nodejs.org/api/events.html#events_class_events_eventemitter)?

File: lib/components/modals/modalviewstack.coffee [hunk 1]
```diff
@@ -13,7 +14,8 @@ module.exports = class KDModalViewStack extends KDObject
   addModal: (modal)->
 
     unless modal instanceof KDModalView
-      return KD.warn "You can only add KDModalView instances to the modal stack."
+      debug 'modal must be an instanceof kd.ModalView'
+      return
```
Comment (usirin) [inline lib/components/modals/modalviewstack.coffee]: `return debug '...'`

**Comment by usirin** [comment]
LGTM

File: lib/index.coffee [hunk 2]
```diff
@@ -4,6 +4,8 @@ module.exports = require './core/kd'
 
 module.exports.extend
 
+  debug                          : require 'debug'
```
Comment (sinan) [inline lib/index.coffee]: why is this necessary?

Comment (sinan) [inline lib/index.coffee]: boilerplate for the other uses?

Comment (tetsuo) [inline lib/index.coffee]: debug should be exposed somehow:

`kd.debug.enable('x:y:z')`

File: lib/core/kd.coffee [hunk 3]
```diff
@@ -12,7 +13,7 @@ getSingleton = (name) ->
   if singletons[name]?
     singletons[name]
   else
-    console.warn "[getSingleton] #{name} doesn't exist"
+    debug "could not found singleton #{name}"
```
Comment (tetsuo) [inline lib/core/kd.coffee]: xxx
